packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -799,7 +827,7 @@ export class Summarizer extends EventEmitter implements ISummarizer {
             // Raise error to parent container.
             this.emit("summarizingError", createSummarizingWarning("Summarizer: HandleSummaryAckFatalError", true));
```

Comment:
I believe we either need to eliminate this type of error or remove try/catch from handleSummaryAcks(). I'd vote for the former.

Reply:
What do you mean eliminate? We can make it a floating promise, but as it is now, this error should never hit already.

Reply:
Take a look at handleSummaryAcks - there is pretty much no way it can throw exception. Essentially we have try/catch around try/catch and both report errors.
If it's expected that handleSummaryAcks is not propagating errors to callers, then .catch(() => {}) with some comment is better

Reply:
Okay I'm fine with that. I was thinking the telemetry log is slightly better, because it goes one of two ways:
1. As intended, it never gets hit, we never see this error in telemetry.
2. Not as intended, it gets hit (something in the underlying code changed!) and we log to telemetry + stop the .

Cons: Cost is this extra 3 lines of code to propagate the error and stop the summarizer client.

Pros: Benefit is no unhandled floating promise: linter is happy & we protect/log against future bugs by stopping the summarizer client. If we don't stop the summarizer client and somehow this gets hit, then this summarizer client is in a bad state since it can't update it's own state on summary ack; I think it would always do safe summary.

Reply:
So to me this is more like "fail fast"/assert as it is now. Curious your thoughts.

Reply:
I'd say I care only about presence (or lack of) handleSummaryAckFatalError in SummarizerStopReason - it feels like it should not be in enum as it should never be hit. Error in telemetry - sure.